### PR TITLE
Home Task 05

### PR DESCRIPTION
**Pull request by isken24**
Home Task 05 is ready for review.

### CODE REVIEW

File: home_task_05.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    def __init__(self):
+        randomizer = randrange(Greetings.number_of_languages)
+        self.greeting = Greetings.GREETINGS[randomizer]
+
+    def get_greeting(self):
+        return self.greeting
+
+
+def create_news():
+    record = News()
```
Comment (dena-den) [inline home_task_05.py:97]: record = News(text_of_news, city)

**Review by dena-den — CHANGES_REQUESTED** [review]

File: home_task_05.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    record.text_of_news = input('Enter text of the news: ')
+    record.city = input('Enter city: ')
+
+    with open('hw_05_result.txt', 'a') as result:
+        result.write(f'{record.header}'
```
[the remark this answers is not in the document]
Comment (isken24) [reply home_task_05.py:102]: Fixed according to request.

File: home_task_05.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+Each new record should be added to the end of file. Commit file in git for review.
+"""
+
+
+from datetime import datetime, date
+from random import randrange
+
+
+class News:
+    header = 'News -------------------------\n'
```
[the remark this answers is not in the document]
Comment (isken24) [reply home_task_05.py:21]: Deleted those variables.

File: home_task_05.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+    @property
+    def city(self):
+        return self.__city
+
+    @city.setter
+    def city(self, city):
+        self.__city = city
+
+    def add_timestamp(self):
```
[the remark this answers is not in the document]
Comment (isken24) [reply home_task_05.py:45]: Got rid of this method.

File: home_task_05.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                     f'\n{record.end_of_record}')
+
+
+def main():
+    while True:
+        try:
+            print('Choose type of record: 1 - News, 2 - Private Ad, 3 - Random Fact, 0 - EXIT PROGRAM')
+            x = int(input('Enter single digit 0-3:'))
+            if x == 0:
+                print('Good buy!')
```
[the remark this answers is not in the document]
Comment (isken24) [reply home_task_05.py:145]: Looks like I'm willing to sell my program :D

File: home_task_05.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                break
+            elif x == 1:
+                create_news()
+                continue
```
[the remark this answers is not in the document]
Comment (isken24) [reply home_task_05.py:149]: Thanks for advise!

File: home_task_05.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+            elif x == 2:
+                create_private_ad()
+                continue
+            elif x == 3:
+                add_greeting()
+                continue
+            else:
+                print(f'Input value must be single digit from 0 to 3!')
```
[the remark this answers is not in the document]
Comment (isken24) [reply home_task_05.py:157]: Didn't manage to do it other way and make it work as it supposed to.

File: home_task_05.py (hunk 1 above)
Comment (isken24) [reply home_task_05.py:97]: Fixed according to request.

File: home_task_05.py [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+
+from datetime import datetime, date
+from random import randrange
+
+
+class News:
+
+    def __init__(self, text_of_news, city):
+        self.publication_date = datetime.now().strftime("%Y/%m/%d %H:%M")
```
Comment (dena-den) [inline home_task_05.py:23]: Why don't you use underscores here, like __publication_date?

File: home_task_05.py [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+    record = Greetings()
+
+    with open('hw_05_result.txt', 'a') as result:
+        result.write(record.create_greeting())
+
+
+def main():
+    while True:
+        try:
+            print('Choose type of record: 1 - News, 2 - Private Ad, 3 - Random Fact, 0 - EXIT PROGRAM')
```
Comment (dena-den) [inline home_task_05.py:145]: 3 - isn't a fact

File: home_task_05.py [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+    @property
+    def expiration_date(self):
+        return self.__expiration_date
+
+    @expiration_date.setter
+    def expiration_date(self, expiration_date):
+        self.__expiration_date = expiration_date
+
+    def number_of_days_left(self):
```
Comment (dena-den) [inline home_task_05.py:73]: better `get_number_of_days_left`

**Review by dena-den — CHANGES_REQUESTED** [review]
Nice. Few improvements in naming.
And another one thought, could you create parent class for all types of publication? And implement in this class common parameters (if you can find so) or at least create method to write publication to file. 
Because there we have 3 types of publications and they should be connected together (using parent class).

File: home_task_05.py (hunk 9 above)
Comment (isken24) [reply home_task_05.py:145]: Fixed

File: home_task_05.py (hunk 8 above)
Comment (dena-den) [reply home_task_05.py]: Good decision, especially if it's more Pythonic.

**Pull request merged**